### PR TITLE
Fix test failure and delete unused code

### LINKED ISSUE #60745: Failure in Conformance test - Daemon set [Serial] should update pod when spec was updated and update strategy is RollingUpdate [Conformance]
Please see : 
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-local-e2e/322

fails just after this line:
```
STEP: Make sure all daemon pods have correct template generation 999
```

Failure was introduced in:
https://github.com/kubernetes/kubernetes/pull/59883

But this was not noticed until it started running in [Conformance] tests.

You can see the same problem in other dashboards too:
https://k8s-testgrid.appspot.com/sig-release-master-upgrade
https://k8s-testgrid.appspot.com/canonical-ubuntudev-k8sdev

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:
`Daemon set [Serial] should update pod when spec was updated and update strategy is RollingUpdate` is broken by recent updates as Template generation isn't supported by apps/v1.DaemonSet anymore

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #60745 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by timothysc** [comment]
There is no issue specified, or link to broken test(s).

**Comment by dims** [comment]
@timothysc please see 60745. i have updated the title to reflect what i am trying to do here (test pull job)

**Comment by dims** [comment]
Looks like my patch fixes the problem in the test (using local-up-cluster) - http://paste.openstack.org/show/692060/

**Review by janetkuo — APPROVED** [review]
Is this fix ready? It looks good to me. Just a nit on log.

**Comment by janetkuo** [comment]
Template generation isn't supported by apps/v1.DaemonSet anymore. The behavior is still preserved in DaemonSet controller for backward compatibility (users can still use it with extensions/v1beta1 endpoint). 

#59883 moved DaemonSet tests to apps/v1 but didn't remove the Template Generation part, which caused the failure this PR tries to fix. 

We should still test template generation to make sure we don't break backward compatibility. That can be done in integration test. cc @crimsonfaith91

File: test/e2e/apps/daemon_set.go [hunk 1]
```diff
@@ -315,8 +314,6 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 	framework.ConformanceIt("should update pod when spec was updated and update strategy is RollingUpdate", func() {
 		label := map[string]string{daemonsetNameLabel: dsName}
 
-		templateGeneration := int64(999)
-		framework.Logf("Creating simple daemon set %s with templateGeneration %d", dsName, templateGeneration)
```
[the remark this answers is not in the document]
Comment (dims) [reply test/e2e/apps/daemon_set.go]: @janetkuo ack, will fix it up in a minute

**Comment by crimsonfaith91** [comment]
@janetkuo Thank you for the information!

**Pull request merged**